### ISSUE TITLE
Failed to open instrumentation library?

### ISSUE THREAD

**Issue by msftsecurityteam** [vrtadmin/moflow#7]
Hi, I have (finally) built Dyninst and got afl-dyninst to build as well.  Like the git instructions say, my libAflDyninst.so is in my current directory but after I run afl-dyninst, it errors out on loading the instrumentation library (it errors saying "The instrumentation library must be in the current working directory.").  Is it possible it's having an issue because the binary I am trying to load is a 32 bit binary on a 64 bit processor?  This particular application distributes only 32 bit binaries and requires the user to have ia32libs type package on their distribution to run it.

**Comment by richinseattle** [comment]
Hi,

I've ran into this issue when DYNINSTAPI_RT_LIB environment variable isn't
set properly.
It should point to libdyninstAPI_RT.so binary as mentioned in the readme.
Does this solve your issue?

I'll add an explicit check and a warning in any case.

Thanks,
Aleks

**Comment by ea** [comment]
Also, am I understanding this correctly that you are trying to instrument an x86 binary on an x86_64 linux, or on a x86 linux, but the processor is x86_64? 
If it's the former, was Dyninst, and afl-dyninst, compiled as 32bit or 64 bit? 
64bit build of Dyninst won't be able to instrument 32bit applications, at least in my tests.

**Comment by msftsecurityteam** [comment]
@richinseattle 

My DYNINSTAPI_RT_LIB is set correctly.

@ea 

Yes, exactly.  amd64 Linux instrumenting an i386 binary.  afl-dyninst built as 64-bit.  I have built Dyninst with something like.... BUILD_RTLIB_32=ON 

I have tried pointing DYNINSTAPI_RT_LIB=/usr/local/lib/libdyninstAPI_RT_m32.so.8.2 but that doesn't help.  I think it's because the libAflDyninst.so is built 64-bit, and cannot be built 32-bit without a completely separate Dyninst build that is 32-bit.  Just wondering if this truly was the problem or if it was something stupid I was doing :)
